2023/03/identifying-vulnerabilities-using-horusec.md

```diff
@@ -0,0 +1,112 @@
+
+<!-- Photo by Pranesh, 2022 -->
+
+## Horusec
+
+[Horusec](https://horusec.io/site/) is an open source tool that orchestrates other security tools and identifies security flaws or vulnerabilities in projects during the development process and puts all results in a database for analysis and generation of metrics.
+
+Currently, Horusec supports - C#, Java, Kotlin, Python, Ruby, Golang, Terraform, Javascript, Typescript, Kubernetes, PHP, C, HTML, JSON, Dart, Elixir, Shell, Nginx, Swift.
+
+It can also be integrated with CI/CD to execute the scan every time a developer raises a merge request.
+
+### Horusec CLI Installation
+
+#### Requirements
+
+- Docker
+- Git
+
+```
+curl -fsSL https://raw.githubusercontent.com/ZupIT/horusec/main/deployments/scripts/install.sh | bash -s latest
+```
+
+### VS Code Extension
+
+Horusec has a [VS code extension](https://docs.horusec.io/docs/extensions/visual-studio-code/) which is super helpful making complete code analysis with a single click.
+
+### Usage
+
+Navigate to application directory:
+
+```
+cd /path/to/project
+```
+
+#### Generate
+
+This command generates a configuration file called `horusec-config.json` which has a set of [customisations](https://docs.horusec.io/docs/cli/commands-and-flags/#global-flags).
+
+```
+horusec generate
+```
+
+#### Start
+
+The start command executes the code scan throughout the repository searching possible vulnerabilities. All the detected vulnerabilities are stored in the outfile JSON file.
+
+```
+horusec start -p . --output-format=json --json-output-file=<filename>.json
+```
+
+### Classification of Vulnerabilities
+
+Horusec may identify vulnerabilities that are not even one. Vulnerabilities have to be classified to avoid some unwanted ones.
+
+##### False positive
+
+Vulnerability found is wrong, because it is accused in a test file or it is not a vulnerability in fact, and is a safe code.
+
+##### Accepted Risk
+
+Vulnerability that was accused, but at the moment, you don't have the option to correct it, so it is classified as an accepted risk to move forward in the daily process.
+
+##### Corrected
+
+Vulnerability that doesn't exist and can be considered as corrected.
+
+##### Vulnerability
+
+A possible security failure found and accused by the analysis.
+
+The hashes to be ignored classified can be added to the configuration file horusec-config.json
+
+```
+"horusecCliFalsePositiveHashes": [
+    HASH1, HASH2
+],
+"horusecCliRiskAcceptHashes": [
+    HASH1, HASH2
+],
+```
+
+#### Ignore files
+
+To ignore a file or certain paths under the directory, add the paths to the configuration file,
+
+```
+"horusecCliFilesOrPathsToIgnore": [
+    PATH1, PATH2
+]
+```
```

Comment:
Likewise this would make more sense to a new reader if you show some actual paths. For example it would help me understand what the paths are relative to: not the filesystem root, I hope, but from where?

Reply:
Updated the code with actual paths.